backend/src/v2/compiler/argocompiler/dag.go

```diff
@@ -72,6 +72,15 @@ func (c *workflowCompiler) DAG(name string, componentSpec *pipelinespec.Componen
 		if err != nil {
 			return err
 		}
+		// Add Parallelism Limit if present
+		parallel := int64(kfpTask.GetIteratorPolicy().GetParallelismLimit())
+		if parallel > 0 {
+			currentParallelism := dag.Parallelism
+			if currentParallelism == nil || parallel > *currentParallelism {
+				dag.Parallelism = &parallel
+			}
+		}
```

Comment:
You could add an else statement here to set a default parallelism limit if none or an invalid one is specified. Something like:

```
else {
    defaultParallelism := int64(1)  // Assuming '1' as a safe default
    dag.Parallelism = &defaultParallelism
    log.Infof("Set default parallelism limit of %d due to invalid input", defaultParallelism)
}

```